build.gradle

```diff
@@ -406,10 +407,92 @@ configure(project(':core')) {
+            File localnetDir = new File(project.rootDir, localnetDirName)
+            localnetDir.mkdirs()
+            File moneroArchiveFile = new File(localnetDir, moneroArchiveFileName)
+            if (!moneroArchiveFile.exists()) {
+                ext.downloadAndVerifyArchive(moneroDownloadUrl, moneroSHA256Hash, moneroArchiveFile)
+            }
+
+            // extract
+            String monerod = 'monerod'
+            String moneroRpc = 'monero-wallet-rpc'
+            if (Os.isFamily(Os.FAMILY_WINDOWS)) {
+                monerod = monerod + '.exe'
+                moneroRpc = moneroRpc + '.exe'
+                ext.extractArchiveZip(moneroArchiveFile, localnetDir)
+            } else {
+                ext.extractArchiveTarGz(moneroArchiveFile, localnetDir)
+            }
+
+            // add the current platform's monero dependencies into the resources folder for installation
+            copy {
+                from "${rootProject.projectDir}/.localnet/${monerod}"
+                into "${project(':core').projectDir}/src/main/resources/bin"
+            }
+            copy {
+                from "${rootProject.projectDir}/.localnet/${moneroRpc}"
+                into "${project(':core').projectDir}/src/main/resources/bin"
+            }
+        }
+
+        ext.downloadAndVerifyArchive = { String archiveURL, String archiveSHA256, File destinationArchiveFile ->
+            println "Downloading ${archiveURL}"
+            ant.get(src: archiveURL, dest: destinationArchiveFile)
+            println 'Download saved to ' + destinationArchiveFile
+
+            println 'Verifying checksum for downloaded binary ...'
+            ant.archiveHash = archiveSHA256
+            ant.checksum(file: destinationArchiveFile, algorithm: 'SHA-256', property: '${archiveHash}', verifyProperty: 'hashMatches')
+            if (ant.properties['hashMatches'] != 'true') {
+                ant.fail('Checksum mismatch: Downloaded JDK binary has a different checksum than expected')
+            }
+            println 'Checksum verified'
+        }
+
+        ext.extractArchiveTarGz = { File tarGzFile, File destinationDir ->
+            println "Extracting tar.gz ${tarGzFile}"
+            // Gradle's tar extraction preserves permissions (crucial for jpackage to function correctly)
+            copy {
+                from tarTree(resources.gzip(tarGzFile))
+                into destinationDir
+            }
+            println "Extracted to ${destinationDir}"
+        }
+
+        ext.extractArchiveZip = { File zipFile, File destinationDir ->
+            println "Extracting zip ${zipFile}..."
+            ant.unzip(src: zipFile, dest: destinationDir)
+            println "Extracted to ${destinationDir}"
+        }
     }
+
     if (osdetector.os != 'windows') processResources.dependsOn havenoDeps // before both test and build
     else println "Haveno dependencies must be downloaded manually on Windows"
```

Comment:
Maybe this line is outdated now that we're using gradle to download and verify dependencies? If unknown we can defer to a future PR for windows.

Reply:
No, windows works fine this is outdated now, deleting.